clusterloader2/pkg/prometheus/prometheus.go

```diff
@@ -183,6 +192,38 @@ func (pc *PrometheusController) exposeKubemarkApiServerMetrics() error {
 	return nil
 }
 
+// runKubemarkNodeExporter creates daemonset with node-exporter in kubemark (not "root"!) cluster.
+func (pc *PrometheusController) runKubemarkNodeExporter() error {
+	klog.Infof("Starting node-exporter on kubemark's master nodes.")
+	kubemarkFramework, err := framework.NewFramework(&pc.clusterLoaderConfig.ClusterConfig, numK8sClients)
+	if err != nil {
+		return err
+	}
+
+	// Validate masters first
+	nodes, err := client.ListNodes(kubemarkFramework.GetClientSets().GetClient())
+	if err != nil {
+		return err
+	}
+
+	numMasters := 0
+	for _, node := range nodes {
+		if system.IsMasterNode(node.Name) {
+			numMasters++
+			if label := node.Labels[nodeExporterNodeLabel]; label != "true" {
+				return fmt.Errorf("Master %q doesn't have required %q label which is required by node-expoter", node.Name, nodeExporterNodeLabel)
+			}
+		}
+	}
+
+	if numMasters == 0 {
+		return fmt.Errorf("node-exporter requires master to be regirstered nodes")
+	}
+
+	return kubemarkFramework.ApplyTemplatedManifests(
+		kubemarkNodeExporterDS, pc.templateMapping, client.Retry(apierrs.IsNotFound))
```

Comment:
I don't think we need to retry IsNotFound here, it's needed for Prometheus CRDs to deflake race conditions, but here we shouldn't expect 404 when creating DaemonSets.